File: 404-sum-of-left-leaves.java
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        return dfs(root,false);
    }
    private int dfs(TreeNode node, boolean flag){
        int ans = 0;
        if(node == null) return 0;
        if(node.left == null && node.right == null) return flag?node.val:0;
        ans += dfs(node.left,true);
        ans += dfs(node.right,false);
        return ans;
    }
}

File: 404-sum-of-left-leaves.py
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def dfs(node,flag):
            ans = 0
            if not node:
                return 0
            elif not node.left and not node.right:
                return node.val if flag else 0
            ans += dfs(node.left, True)
            ans += dfs(node.right, False)
            return ans
        
        return dfs(root,False)